src/shave.cpp

```diff
@@ -350,7 +351,26 @@ void filterFeatures(vtzero::tile_builder* finalvt,
                 feature_builder.set_id(feature.id());
             }
             feature_builder.set_geometry(feature.geometry());
+
+            std::string property_filter_type = property_filter.type;
+            std::vector<std::string> properties = property_filter.values;
+            auto keytable = layer.key_table();
+
+            bool needAllProperties = property_filter_type == "all";
```

Comment:
I think this calculation could be moved up in scope, for best efficiency, above `layer.for_each_feature`, since it does not change per feature.

Reply:
👍 Agree, To the extreme, from small details.